source/scripts/script.js

```diff
@@ -87,193 +98,45 @@ function search() {
-        bookmarkIcon.src = "./img/icons/bookmark-filled.svg";
-    }
-    else {
-        bookmarkIcon.src = "./img/icons/bookmark-empty.svg";
-    }
-}
-
-function updateSettings() {
-    const dietaryRestrictionList = [];
-    // Get all the inputs under the div
-    const dietaryContainerElements = document.getElementById('dietary-container').elements;
-    for (let i = 0; i < dietaryContainerElements.length; i++) {
-        // If a checkbox is checked, then add it to our list
-        const inputElement = dietaryContainerElements[i];
-        if (inputElement.checked) {
-            dietaryRestrictionList.push(inputElement.value);
-        }
-    }
-
-    const intolerancesRestrictionsList = [];
-    const intolerancesContainerElements = document.getElementById("intolerances-container");
-    for (let i = 0; i < intolerancesContainerElements.length; i++) {
-        // If a checkbox is checked, then add it to our list
-        const inputElement = intolerancesContainerElements[i];
-        if (inputElement.checked) {
-            intolerancesRestrictionsList.push(inputElement.value);
-        }
-    }
-
-    // Add lists to local storage
-    localStorage.setItem("dietaryRestrictions", JSON.stringify(dietaryRestrictionList));
-    localStorage.setItem("intolerancesRestrictions", JSON.stringify(intolerancesRestrictionsList));
-
-    // TODO: add confirmation message in HTML (alert is temporary)
-    alert("your preferences have been updated");
-}
+// main.js
 
 function createRecipeCards() {
     const recipeCardContainer = document.getElementById('recipe-card-container');
     for (let i = 0; i < recipeData.length; i++) {
         const element = document.createElement('recipe-card');
         element.data = recipeData[i];
         document.querySelector("recipe-page").data = recipeData[i];
+
+        const page = recipeData[i]["title"];
+
+        router.addPage(page, function () {
+            hideHome();
+            hideRecipeCards();
+            showRecipePage();
+            document.querySelector("recipe-page").data = recipeData[i];
+            checkBookMark(recipeData[i]);
+        });
+
         recipeCardContainer.appendChild(element);
-        bindRecipeCard(element);
+        bindRecipeCard(element, page);
     }
 }
 
-function bindRecipeCard(recipeCard) {
-    recipeCard.addEventListener('click', event => {
-        console.log("Recipe card has been clicked");
-        const recipePageContainer = document.querySelector("#recipe-page-container");
-        hideHome();
-        hideRecipeCards();
-        recipePageContainer.classList.remove("hidden");
+function bindRecipeCard(recipeCard, pageName) {
+    recipeCard.addEventListener('click', e => {
+        if (e.path[0].nodeName == 'A') return;
+        router.navigate(pageName);
     });
 }
 
+function bindPopstate() {
+    window.addEventListener("popstate", (event) => {
+        if (event.state != null)
+            router.navigate(event.state, true);
```

Comment:
Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8422335)
```suggestion
            {router.navigate(event.state, true);}
```